napari/resources/_icons.py

```diff
@@ -0,0 +1,53 @@
+import re
+from functools import lru_cache
+from pathlib import Path
+
+ICON_PATH = (Path(__file__).parent / 'icons').resolve()
+ICONS = {x.stem: x for x in ICON_PATH.iterdir()}
+
+
+def get_icon_path(name: str) -> str:
+    """Return path to an SVG in the theme icons."""
+    icon = ICONS.get(name)
+    if not icon:
+        raise ValueError(
+            f"unrecognized icon name: {name!r}. Known names: {ICONS}"
+        )
+    return str(icon)
+
+
+svg_elem = re.compile(r'(<svg[^>]*>)')
+svg_style = """<style type="text/css">
+path {{fill: {0}; opacity: {1};}}
+polygon {{fill: {0}; opacity: {1};}}
+circle {{fill: {0}; opacity: {1};}}
+rect {{fill: {0}; opacity: {1};}}
+</style>"""
+
+
+@lru_cache()
+def get_raw_svg(path: str) -> str:
+    """Get and cached SVG XML.
+
+    Raises
+    ------
+    ValueError
+        If the path exists but does not contain valid SVG data.
+    """
+    with open(path) as f:
+        xml = f.read()
+        if not svg_elem.search(xml):
```

Comment:
is the regex overkill ? You already match for `</svg>` in get_colorized_svg ...

Reply:
the `if '</svg>'` in `get_colorized_svg` is slightly different... just to make sure that the string that was passed wasn't already an svg xml string. the `svg_elem`  regex is more important for the `svg_elem.sub` call at the end of `get_colorized_svg`... where I specifically need to find the second `>` in the opening svg tag.  so, I don't need it in `get_raw_svg`, but if it fails there, it will fail in `get_colorized_svg`... thoughts?

Reply:
No particular thoughts; I just thought  it might have been a leftover from refactor, and leave a comment. It can still be useful for future reference, or other reviewers. 

This is neither a hot spot, nor a user exposed function; so I don't think it's worth nitpicking, or chasing details.

Reply:
I moved it down into the `get_colorized_svg` where it really belongs. thanks!